asv_bench/benchmarks/pandas/frame_methods.py

```diff
@@ -0,0 +1,292 @@
+import string
+import warnings
+
+import numpy as np
+
+from modin.pandas import (
+    DataFrame,
+    MultiIndex,
+    NaT,
+    Series,
+    date_range,
+    isnull,
+    period_range,
+    timedelta_range,
+)
+
+import pandas._testing as tm
+
+
+class Reindex:
+    def setup(self):
+        N = 10**3
+        self.df = DataFrame(np.random.randn(N * 10, N))
+        self.idx = np.arange(4 * N, 7 * N)
+        self.idx_cols = np.random.randint(0, N, N)
+        self.df2 = DataFrame(
+            {
+                c: {
+                    0: np.random.randint(0, 2, N).astype(np.bool_),
+                    1: np.random.randint(0, N, N).astype(np.int16),
+                    2: np.random.randint(0, N, N).astype(np.int32),
+                    3: np.random.randint(0, N, N).astype(np.int64),
+                }[np.random.randint(0, 4)]
+                for c in range(N)
+            }
+        )
+
+    def time_reindex_axis0(self):
+        self.df.reindex(self.idx)
+
+    def time_reindex_axis1(self):
+        self.df.reindex(columns=self.idx_cols)
+
+    def time_reindex_axis1_missing(self):
+        self.df.reindex(columns=self.idx)
+
+    def time_reindex_both_axes(self):
+        self.df.reindex(index=self.idx, columns=self.idx_cols)
+
+    def time_reindex_upcast(self):
+        self.df2.reindex(np.random.permutation(range(1200)))
+
+
+class Rename:
+    def setup(self):
+        N = 10**3
+        self.df = DataFrame(np.random.randn(N * 10, N))
+        self.idx = np.arange(4 * N, 7 * N)
+        self.dict_idx = {k: k for k in self.idx}
+        self.df2 = DataFrame(
+            {
+                c: {
+                    0: np.random.randint(0, 2, N).astype(np.bool_),
+                    1: np.random.randint(0, N, N).astype(np.int16),
+                    2: np.random.randint(0, N, N).astype(np.int32),
+                    3: np.random.randint(0, N, N).astype(np.int64),
+                }[np.random.randint(0, 4)]
+                for c in range(N)
+            }
+        )
+
+    def time_rename_single(self):
+        self.df.rename({0: 0})
+
+    def time_rename_axis0(self):
+        self.df.rename(self.dict_idx)
+
+    def time_rename_axis1(self):
+        self.df.rename(columns=self.dict_idx)
+
+    def time_rename_both_axes(self):
+        self.df.rename(index=self.dict_idx, columns=self.dict_idx)
+
+    def time_dict_rename_both_axes(self):
+        self.df.rename(index=self.dict_idx, columns=self.dict_idx)
+
+
+class Repr:
+    def setup(self):
+        nrows = 10000
+        data = np.random.randn(nrows, 10)
+        arrays = np.tile(np.random.randn(3, nrows // 100), 100)
+        idx = MultiIndex.from_arrays(arrays)
+        self.df3 = DataFrame(data, index=idx)
+        self.df4 = DataFrame(data, index=np.random.randn(nrows))
+        self.df_tall = DataFrame(np.random.randn(nrows, 10))
+        self.df_wide = DataFrame(np.random.randn(10, nrows))
+
+    def time_html_repr_trunc_mi(self):
```

Comment:
what exactly do these do? I feel like we don't want to benchmark them, at least repr_html

Reply:
removed